mmlearn/modules/encoders/vision.py

```diff
@@ -0,0 +1,151 @@
+        Whether to freeze the layer norm.
+    peft_config : Optional[PeftConfig], default=None
+        The PEFT configuration.
+    model_kwargs : Optional[Dict[str, Any]], default=None
+        Additional keyword arguments for the model.
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        projection_dim: int = 768,
+        pretrained: bool = True,
+        freeze_layers: Union[int, float, List[int], bool] = False,
+        freeze_layer_norm: bool = True,
+        peft_config: Optional[PeftConfig] = None,
+        model_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """Initialize the Vision Transformer model."""
+        super().__init__()
+        if model_kwargs is None:
+            model_kwargs = {}
+
+        model: nn.Module = timm.create_model(
+            model_name,
+            pretrained=pretrained,
+            num_classes=projection_dim,
+            **model_kwargs,
+        )
+        assert isinstance(model, VisionTransformer), (
+            f"Model {model_name} is not a Vision Transformer. "
+            "Please provide a model name that corresponds to a Vision Transformer."
+        )
+
+        if isinstance(freeze_layers, bool) and freeze_layers:
+            for name, param in model.named_parameters():
+                param.requires_grad = (
+                    (not freeze_layer_norm) if "norm" in name else False
+                )
+
+        modules = [model.patch_embed, *model.blocks, model.norm]
+        if isinstance(freeze_layers, float):
+            freeze_layers = int(freeze_layers * len(modules))
+        if isinstance(freeze_layers, int):
+            freeze_layers = list(range(freeze_layers))
+
+        if isinstance(freeze_layers, list):
+            for idx, module in enumerate(modules):
+                if idx in freeze_layers:
+                    for name, param in module.named_parameters():
+                        param.requires_grad = (
+                            (not freeze_layer_norm) if "norm" in name else False
+                        )
+
+        if peft_config is not None:
+            model = hf_utils._wrap_peft_model(model, peft_config)
+
+        self.model = model
+
+    def forward(self, inputs: Dict[Union[str, Modality], Any]) -> BaseModelOutput:
+        """Run the forward pass.
+
+        Parameters
+        ----------
+        inputs : Dict[str | Modality, Any]
+            The input data. The `image` will be expected under the `Modalities.RGB` key.
+
+        Returns
+        -------
+        BaseModelOutput
+            The output of the model.
+        """
+        x = inputs[Modalities.RGB]
+        x = self.model.forward_features(x)
+
+        # Separate the class token and patch embeddings
+        cls_token = x[:, 0]
+        patch_embeddings = x[:, 1:]
+
+        return BaseModelOutput(
+            last_hidden_state=patch_embeddings,
+            pooler_output=cls_token,
+            hidden_states=None,
```

Comment:
This should return the output of the intermediate layers to match the style of huggingface encoders.

Reply:
I mean `hidden_states` should not be `None`

Reply:
Fixed